mozilla/donate-wagtail
# Add support for selecting a different merchant account for Paypal based on donation amount

Fixes #211.

@alanmoo there are a couple of things to note here:

1. I have copied the logic from [the current site](https://github.com/mozilla/donate.mozilla.org/blob/9823bb7a671ede8a267e850faf00ff0f767db601/server/lib/paypal-account-switcher.js) to determine which account to use. The basic logic here is:

      - Assume micro payments come with a transaction fee of 6%, plus a fixed fee which is currency specific.
      - Assume macro payments come with a transaction fee of 2.9%, plus a fixed fee which is currency specific.

    Paypal's [published fees](https://www.paypal.com/us/webapps/mpp/merchant-fees#fixed-fees-standard) are slightly more complex in that they vary depending on whether the payment is coming from within the US. The micro transaction fee is either 5% or 6.5% depending on this.

    I'm assuming that the current calculations are basically approximating an average of these, but let me know if you want to refine further. I think it is also worth someone going through and checking that the fixed fees for each currency (defined in `payments/constants.py`) are still correct.

2. I can't find any information from Paypal about charges for subscriptions - so have assumed that the same rates apply, and on that basis subscriptions are created using the micro/macro account using the same logic as above, based on the amount of the monthly subscription.

To test this:

1. On the review app, define a new env var `BRAINTREE_MERCHANT_ACCOUNTS_PAYPAL_MICRO`, which is in the same format as `BRAINTREE_MERCHANT_ACCOUNTS`. It specifies which merchant account to use for each currency, for micro transactions.

2. Make small ($1) and large ($20) payments, and verify that these are put through the correct Paypal accounts.

3. Set up small ($1) and large ($20) monthly subscriptions, and verify that these are put through the correct Paypal accounts.
